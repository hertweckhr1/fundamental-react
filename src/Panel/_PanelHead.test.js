import { mount } from 'enzyme';
import PanelHead from './_PanelHead';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<Panel />', () => {
    const panelHead = (
        <PanelHead
            description='Panel Description'
            title={'Panel Header with Actions'} />
    );

    test('create panel Head', () => {
        // create panel hed
        let component = renderer.create(panelHead);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the PanelHead component', () => {
            const element = mount(<PanelHead data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
