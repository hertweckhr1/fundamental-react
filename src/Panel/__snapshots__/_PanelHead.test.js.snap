// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Panel /> create panel Head 1`] = `
<div
  className="fd-panel__head"
>
  <h1
    className="fd-panel__title"
  >
    Panel Header with Actions
  </h1>
  <p
    className="fd-panel__description"
  >
    Panel Description
  </p>
</div>
`;
